### DISCUSSION TITLE
Lucene search index not taking widget contents

### DISCUSSION THREAD

**Discussion by momentum-tunis** [OrchardCMS/OrchardCore#11289] [Q&A] [answered]
Hi,

It is the first time that use the Lucene fulltext search feature and I didn't get it how to search inside parts.

If I use the blog theme I can search for text inside blog posts, I can also retrieve pages when searching for their titles.

Now if add a widget inside a page (content type Page with a Flow part), and if I add a Paragraph widget inside, with a word "foobar", how should I setup the recipe to let me search for "foobar", in other terms how to add the FlowPart widgets contents to the Lucene index.

I checked "Include this element in the index" everywhere and still not working. I think that I should use the "Use custom full-text" in the Page content type definition, but I don't know how to add the flowparts whith it.

Any Ideas? Thanks in advance

**Comment by momentum-tunis** [comment]
My issue is exactly what is described here:
https://github.com/OrchardCMS/OrchardCore/issues/2675
It is not working even with checking "Include this element index" at every level

**Comment by momentum-tunis** [comment]
More details:
- only the first item in a BagPart is added to the index
- flowpart items are not indexed even if checking "Include this element in the index"

**Comment by momentum-tunis** [comment, marked as answer]
I finally succed to get this to work. For anyone having the same issue to search the content inside widgets of a flowpart, you should specify the liquid template in "Use custom full-text" in the parent content type definition, in my case "Page":

{{ Model.Content.FlowPart.Widgets | full_text_aspect }}

![image](https://user-images.githubusercontent.com/58601327/156742196-50a08642-feb8-462f-979c-25435f49a4a6.png)

Do not forget also to check "Include this element in the index" in child elements content type definition.

I think this is something that should be enabled by default, or at least well explained with examples in the documentation here
https://docs.orchardcore.net/en/dev/docs/guides/implement-fulltext-search/

**Reply by Skrypt** [reply]
Please create a Pull Request if you want to contribute to the documentation and share your experience with everyone. I believe that this is something that @deanmarcussen added in OC and I'm not sure if it has been documented.

**Reply by momentum-tunis** [reply]
https://github.com/OrchardCMS/OrchardCore/pull/11320

It is my first pr in the project, hope it follows the guidelines
